src/include/btree.i

```diff
@@ -489,6 +489,8 @@ __wt_page_only_modify_set(WT_SESSION_IMPL *session, WT_PAGE *page)
     if (page->modify->page_state < WT_PAGE_DIRTY &&
       __wt_atomic_add32(&page->modify->page_state, 1) == WT_PAGE_DIRTY_FIRST) {
         __wt_cache_dirty_incr(session, page);
+        if (page->read_gen == WT_READGEN_WONT_NEED)
+            __wt_cache_read_gen_new(session, page);
```

Comment:
Did you consider using __wt_cache_read_gen_bump instead? I'm really only asking because you said there is a 10% regression remaining, and I wonder if setting the read generation to current rather than middle of the road would help.